### DISCUSSION TITLE
Driver for linux?

### DISCUSSION THREAD

**Discussion by shorton3** [carlmontanari/scrapli#251] [Q&A] [answered]
Is there a scrapli driver for just plain linux?   (Actually, I'm looking for a scrapli driver for F5 network devices, but they are basically just linux vm's, so I'm thinking a linux driver would work...?) Thank you

**Comment by carlmontanari** [comment, marked as answer]
For just linux-y things there are probably better options than scrapli (ansible for example), but if you want to you would use the `GenericDriver`. The generic one has no concept of privilege levels and other network-y things. Docs [here](https://carlmontanari.github.io/scrapli/user_guide/advanced_usage/#using-the-genericdriver).

hth

**Comment by lastorel** [comment]
I'm working on driver for linux (NetworkDriver based) and i'm stuck in privelege management (on sudo escalate prompt). My model contains only "exec" and "root" levels.

Scenario example for `self.acquire_priv("root")`:  
write: `'sudo -s'`  
read : `b'\n\x1b[?2004l\x1b[?2004hcisco@device5:~$ sudo -s'`  
write: `'\n'`  
read : `b'\n\x1b[?2004l[sudo] password for cisco: '` (cisco is username)  
operation timed out, closing connection

I've tried many regexpressions and tested validation through online services like regex101.  
even `escalate_prompt=r"^.*?:\s?$"` can't help

**Reply by lastorel** [reply]
Sorry, my fault. I forgot about `escalate_auth` again. Now it works!

**Comment by lastorel** [comment]
Ansible is "heavy" and slow. Netmiko is slow. Paramiko is hard for create new driver. Your lib is the best choice at the moment.
